lib/elixir/lib/range.ex

```diff
@@ -47,6 +48,12 @@ defmodule Range do
       false
 
   """
+  # @spec range?(term) :: boolean
+  @spec range?(arg) :: false when arg: %Range{first: nil, last: nil}
+  @spec range?(arg) :: true when arg: %Range{}
+  @spec range?(term) :: false 
```

Comment:
This breaks things like Foo.an_integer..3.

As I said in the other email, we cannot check it at compile time. Whatever
you come up with, there will be a false negative or a false positive. Let's
please leave the code as it was and I will try to come up with a better
mechanism, but it may not be possible.


-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D